File: P1 2016.2/Questao(2)/python/entities/produtos/Carro.py
from .Ligavel import Ligavel
from .Brinquedo import Brinquedo


class Carro(Brinquedo, Ligavel):
    def __init__(self, preco: float, id: int, placa: str):
        super().__init__(preco, id)
        self.placa = placa

    def ligar(self):
        print("Girando a chave do carro")

    def __str__(self) -> str:
        return f"Preco: {self.preco}, ID:{self.id}, Placa:{self.placa}"

File: P1 2016.2/Questao(2)/java/entities/produtos/Carro.java
package entities.produtos;

public class Carro extends Brinquedo implements Ligavel {
    private String placa;

    public Carro(double preco, int id, String placa) {
        super(preco, id);
        this.placa = placa;
    }

    public String getPlaca() {
        return placa;
    }

    public void ligar() {
        System.out.println("Girando a chave do carro");
    }
}
